server/backend/api/app/routers/router_application.py

```diff
@@ -24,8 +24,15 @@ def get_applications(
     """
     LOGGER.debug(f"running router ... {db}")
     query_data = crud_application.get_applications_by_granted_apps(db, access_roles)
-    if len(query_data) == 0:
-        response.status_code = 204
+
+    # from fastapi v0.79.0, setting status_code to 204, 304, or any code below 200 (1xx) will remove the body from the response
+    # so when return response with status code 204, response will have no content return to user
+    # if we want to use status code 204 for empty case, we might want to apply to other methods as well
+    # and it will impact some test cases, casue the response cannot be conver to json in empty case
+
+    # if len(query_data) == 0:
+    #     response.status_code = 204
+
```

Comment:
This is the only method set status code to 204 when no applications to return. I comment this out, and let it still return an empty string. But if we still want to use 204 with an empty body, we might want to apply to other methods as well to keep consistency and update the tests.